core/src/main/java/com/oath/oak/Handle.java

```diff
@@ -51,11 +51,11 @@ boolean remove(MemoryManager memoryManager) {
         return true;
     }
 
-    void put(V newVal, OakSerializer<V> serializer, MemoryManager memoryManager) {
+    boolean put(V newVal, OakSerializer<V> serializer, MemoryManager memoryManager) {
         writeLock.lock();
```

Comment:
If not to make Handle synchronization pluggable, let us at least encapsulate the Handle lock into some start/end wrap...

Reply:
I agree that we should revise our implementation to support pluggable Handle implementations.
I suggest we do it in a different PR, to keep the code review manageable.